### PR TITLE
[DOC] Update classification API docs

### PR DESCRIPTION
**Pull request by itsdivya1309**
#### Reference Issues/PRs
Closes: #1497 

#### What does this implement/fix? Explain your changes.

Adds classifiers to the existing list and arranges them in lexicographic order.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

File: docs/api_reference/classification.rst [hunk 1]
```diff
@@ -111,11 +113,14 @@ Interval-based
 
     CanonicalIntervalForestClassifier
```
Comment (baraline) [inline docs/api_reference/classification.rst:114]: Aren't we missing RSTSF as classifier in this section ?

File: docs/api_reference/classification.rst [hunk 2]
```diff
@@ -111,11 +113,14 @@ Interval-based
...
     DrCIFClassifier
+    IntervalForestClassifier
+    QUANTClassifier
+    RandomIntervalClassifier
     RandomIntervalSpectralEnsembleClassifier
+    RDSTClassifier
```
Comment (baraline) [inline docs/api_reference/classification.rst:120]: RDSTClassifier is a shapelet classifier

**Review by baraline — CHANGES_REQUESTED** [review]
Hi @itsdivya1309 Thanks for the PR. Mostly looking good, see the two comments, also, you can add the TEASER classifier in the early classification section.

File: docs/api_reference/classification.rst (hunk 1 above)
Comment (itsdivya1309) [reply docs/api_reference/classification.rst:114]: Yeah, I missed that, thanks.

File: docs/api_reference/classification.rst (hunk 2 above)
Comment (itsdivya1309) [reply docs/api_reference/classification.rst]: Yup, thanks for pointing this out. I've corrected this.

**Review by baraline — APPROVED** [review]
Everything looks good to me !

**Pull request merged**